src/popup/components/FormSelectOptionsHeader.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <div class="form-select-options-header">
+    <span class="text-heading-3 title">{{ title }}</span>
+    <InputSearch
+      v-if="options.length > SHOW_SEARCH_THRESHOLD"
+      :value="searchTerm"
+      :placeholder="searchPlaceholder || $t('common.search')"
+      class="search-field"
+      @input="filterOptions"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType, ref } from '@vue/composition-api';
+import type { TranslateResult } from 'vue-i18n';
+import type { IFormSelectOption } from '../../types';
+import InputSearch from './InputSearch.vue';
+
+const SHOW_SEARCH_THRESHOLD = 3;
+
+export default defineComponent({
+  components: { InputSearch },
+  props: {
+    value: { type: String, default: '' },
+    title: { type: String, default: '' },
+    searchPlaceholder: { type: String as PropType<String | TranslateResult>, default: '' },
+    options: { type: Array as PropType<IFormSelectOption[]>, default: () => [] },
+  },
+  setup(props, { emit }) {
+    const searchTerm = ref('');
+    function filterOptions(searchPhrase: string) {
```

Comment:
This component should not do the actual filtering. Instead it should emit the search phrase and the parent component should react on this.